packages/core/src/renderables/Input.ts

```diff
@@ -294,6 +294,16 @@ export class InputRenderable extends Renderable {
     this.needsUpdate()
   }
 
+  public updateFromLayout(): void {
+    super.updateFromLayout()
+    this.updateCursorPosition()
+  }
+
+  protected onResize(width: number, height: number): void {
+    super.onResize(width, height)
+    this.updateCursorPosition()
+  }
```

Comment:
This is the only significant part of this PR. Other changes are prettier changes.

I could check if the layout has changed before calling `updateCursorPosition` to avoid unnecessary calls, but I couldn't find a reliable way to do it. Not sure about the performance impact of calling `updateCursorPosition`.

Reply:
onResize is only called when the actual size of the renderable in the layout changes. We could add something similar in the base Renderable like `protected onLayoutChanged` that can be overridden by subclasses, just like onResize. And could be called by `updateFromLayout` when either size OR position change. Wdyt? This works as well though. `updateFromLayout` is not supposed to be called by subclasses, even though performance impact would be negligible. Would maybe prefix that with an underscore. I don't see a better way to handle tree update methods atm.